cpp/src/io/orc/orc_reader_impl.cu

```diff
@@ -72,10 +72,12 @@ constexpr std::pair<gdf_dtype, gdf_dtype_extra_info> to_dtype(
       return std::make_pair(GDF_STRING, gdf_dtype_extra_info{TIME_UNIT_NONE});
     case orc::TIMESTAMP:
       // There isn't a GDF_TIMESTAMP -> np.dtype mapping so use np.datetime64
-      return std::make_pair(GDF_DATE64, gdf_dtype_extra_info{TIME_UNIT_ms});
+      return (use_np_dtypes) ? std::make_pair(GDF_DATE64, gdf_dtype_extra_info{TIME_UNIT_ms})
```

Comment:
Maybe I'm confused here but isn't `GDF_TIMESTAMP` the numpy-compatible type?  Now you're returning `GDF_DATE64` when numpy-compatible is enabled.

Reply:
No. GDF_TIMESTAMP and GDF_DATE32 are not compatible with numpy, so we were previously returning GDF_DATE64 to be numpy-compatible (I'm not entirely sure why DATE64 is fine but TIMESTAMP isn't, since the data values are identical, but it's presumably something about how gdf types are mapped to numpy dtypes)
use_np_dtypes=false is the new path (use_np_dtypes=true remains the default).

Reply:
Okay, I see.  Yeah this whole compatibility is a bit fuzzy to me :-).